### PR TITLE
Have the DriverTool, AeronStat, LossStat, and ErrorStat use the C API 

### PR DESCRIPTION
**Pull request by mikeb01**

### CODE REVIEW

File: aeron-client/src/main/c/aeron_cnc.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef AERON_AERON_CNC_H
+#define AERON_AERON_CNC_H
+
+#endif //AERON_AERON_CNC_H
```
Comment (mjpt777) [inline aeron-client/src/main/c/aeron_cnc.h:20]: Why have an empty header?

File: aeron-client/src/main/c/aeronc.c [hunk 2]
```diff
@@ -39,72 +39,47 @@ int aeron_client_connect_to_driver(aeron_mapped_file_t *cnc_mmap, aeron_context_
...
-            aeron_micro_sleep(1000);
-            continue;
+            case AERON_CNC_LOAD_AWAIT_FILE:
+                if (context->epoch_clock() > deadline_ms)
+                {
+                    aeron_set_err(AERON_CLIENT_ERROR_DRIVER_TIMEOUT, "CnC file not created: %s", filename);
+                    return -1;
+                }
+                aeron_micro_sleep(16 * 1000);
+                continue;
```
Comment (mjpt777) [inline aeron-client/src/main/c/aeronc.c:60]: I had to check what continue in a switch did. Not seen that before.

File: aeron-driver/src/test/c/aeron_c_cnc_test.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+    aeron_socket_t fd = bindToLocalUdpSocket(24325);
+    ASSERT_LT(0, fd);
+
+    aeron_async_add_subscription_t *async;
+    ASSERT_EQ(aeron_async_add_subscription(
+        &async, m_aeron, PUB_URI, STREAM_ID, nullptr, nullptr, nullptr, nullptr), 0) << aeron_errmsg();
+    ASSERT_EQ(nullptr, awaitSubscriptionOrError(async));
+
+    filter = {
```
Comment (mjpt777) [inline aeron-driver/src/test/c/aeron_c_cnc_test.cpp:187]: Formatting.

File: aeron-driver/src/test/c/aeron_c_cnc_test.cpp [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+}
+
+TEST_F(CncTest, shouldGetCountersAndLossReport)
+{
+    ASSERT_TRUE(connect());
+    ASSERT_TRUE(openAeronCnc()) << aeron_errmsg();
+    int lossCallbackCounter = 0;
+
+    aeron_counters_reader_t *counters = aeron_cnc_counters_reader(m_cnc);
+    CounterIdFilter filter = {
```
Comment (mjpt777) [inline aeron-driver/src/test/c/aeron_c_cnc_test.cpp:209]: Formatting.

File: aeron-samples/src/main/c/driver_tool.c [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+static void aeron_driver_tool_print_error_and_usage(const char *message)
+{
+    fprintf(stderr, "%s\n%s", message, aeron_driver_tool_usage());
+}
+
+int main(int argc, char **argv)
+{
+    char default_directory[AERON_MAX_PATH];
+    aeron_default_path(default_directory, AERON_MAX_PATH);
+    aeron_driver_tool_settings_t settings = {
```
Comment (mjpt777) [inline aeron-samples/src/main/c/driver_tool.c:60]: Formatting.

File: aeron-samples/src/main/c/error_stat.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        (int)error_length,
+        error);
+}
+
+
+int main(int argc, char **argv)
+{
+    char default_directory[AERON_MAX_PATH];
+    aeron_default_path(default_directory, AERON_MAX_PATH);
+    aeron_error_stat_settings_t settings = {
```
Comment (mjpt777) [inline aeron-samples/src/main/c/error_stat.c:84]: Formatting.

File: aeron-samples/src/main/c/loss_stat.c [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        (int)source_length,
+        source);
+}
+
+
+int main(int argc, char **argv)
+{
+    char default_directory[AERON_MAX_PATH];
+    aeron_default_path(default_directory, AERON_MAX_PATH);
+    aeron_loss_stat_settings_t settings = {
```
Comment (mjpt777) [inline aeron-samples/src/main/c/loss_stat.c:87]: Formatting.

**Review by mjpt777 — CHANGES_REQUESTED** [review]
LGTM with minor changes.

File: aeron-client/src/main/c/aeronc.c (hunk 2 above)
Comment (mikeb01) [reply aeron-client/src/main/c/aeronc.c:60]: I think that it is the first time I'd used it, but it seemed to be the cleanest way to implement it.

File: aeron-client/src/main/c/aeron_cnc.h (hunk 1 above)
Comment (mikeb01) [reply aeron-client/src/main/c/aeron_cnc.h:20]: It has some definitions, which I later removed.  Will delete.

**Pull request merged**